kore/src/Kore/Step/Simplification/Equals.hs

```diff
@@ -39,7 +39,7 @@ import qualified Kore.Step.Simplification.And as And
                  ( simplifyEvaluated )
 import qualified Kore.Step.Simplification.AndTerms as AndTerms
                  ( termEquals )
-import qualified Kore.Step.Simplification.Ceil as Ceil
+import {-# SOURCE #-} qualified Kore.Step.Simplification.Ceil as Ceil
```

Comment:
The proliferation of cyclic dependencies is a burden I'd like to avoid. Without going into all the reasons why, I think that one of the next things we should look at is paying off this technical debt.

Reply:
LGTM++